openedx/openedx-platform
# feat: add video_images_upload_enabled api so new video editor knows val

## Description

This PR adds a new API, `video_images_upload_enaabled`, that allows the `course-authoring` MFE to know that value of the `videos.video_image_upload_enabled` waffle switch. This value is needed to rendered the correct thumbnail experience for users. If the switched is disabled, the user should not have the ability to edit the video thumbnail. If the switched is enabled, the user should have the ability to edit the video thumbnail.

This change impacts "Developers"

## Supporting information

JIRA Ticket: [TNL-9821](https://2u-internal.atlassian.net/browse/TNL-9821)

## Testing instructions

1. Navigate to linked PR from edx/frontend-lib-content-components and switch to the `KristinAoki/thumbnail-widget` branch
2. Navigate to [Django Admin](http://localhost:18010/admin/waffle/flag) and turn on the `new_core_editors.use_new_video_editor` waffle flag
3. Navigate to [Django Admin](http://localhost:18010/admin/waffle/switch) and turn on the `videos.video_image_upload_enabled` waffle switch
4. Open a course in Studio
7. Navigate to the course outline
8. Open a unit to edit
9. Add a "Video" block
10. Scroll to the "Thumbnail" section
11. Check that there is a delete icon (trashcan) next to thumbnail
12. Navigate to [Django Admin](http://localhost:18010/admin/waffle/switch) and turn off the `videos.video_image_upload_enabled` waffle switch
13. Navigate back to "Video" block
14. Check that there is not a delete icon (trashcan) next to thumbnail

## Deadline

None

## Other information

frontend-lib-content-components [PR: 121](https://github.com/edx/frontend-lib-content-components/pull/121) is dependent on this change
